### PR TITLE
Preparations for CFOpenDirectory

### PR DESCRIPTION
**Pull request by CuriousTommy**
This adds some of the files needed for CFOpenDirectory. Keep in mind that the assigned values are just guesses. I don't know what the true values are. I am also not sure if these variables are even in the right spot.

Let me know if you need me to fix anything.

### CODE REVIEW

File: src/opendirectory/include/opendirectory/odutils.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+extern const int eODCustomCallSearchSetPolicyAutomatic;
+extern const int eODCustomCallSearchSetPolicyLocalOnly;
+extern const int eODCustomCallSearchSetPolicyCustom;
+
+void *schema_get_value_at_index(CFArrayRef value, int index);
+void schema_deconstruct_result(CFDictionaryRef dict, CFStringRef *function_name, CFArrayRef *response);
+CFDictionaryRef schema_construct_requestv(CFStringRef function_name, CFIndex total_params, va_list ap);
+CFDictionaryRef schema_construct_request(CFStringRef function_name, CFIndex total_params, ...);
+
+#endif
```
Comment (ahyattdev) [inline src/opendirectory/include/opendirectory/odutils.h]: Could you make sure your files all end with newlines?

File: src/opendirectory/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+project(opendirectory_internal)
+
+add_library(opendirectory_internal src/odutils.c src/odconstants.c)
```
Comment (ahyattdev) [inline src/opendirectory/CMakeLists.txt]: What is `opendirectory_internal` going to be used for?

File: src/opendirectory/include/opendirectory/odutils.h (hunk 1 above)
Comment (CuriousTommy) [reply src/opendirectory/include/opendirectory/odutils.h]: I added the newlines

File: src/opendirectory/CMakeLists.txt (hunk 2 above)
Comment (CuriousTommy) [reply src/opendirectory/CMakeLists.txt]: It is going to be used for CFOpenDirectory. In particular, these files.

- [odxpc.c](https://github.com/darlinghq/darling-opendirectory/blob/c71415e716a8cba3e1539236b49de8abb3de8fe7/Core/odxpc.c#L6)
- [transaction.c](https://github.com/darlinghq/darling-opendirectory/blob/c71415e716a8cba3e1539236b49de8abb3de8fe7/Core/transaction.c#L28)
- [CFODQuery.c](https://github.com/darlinghq/darling-opendirectory/blob/c71415e716a8cba3e1539236b49de8abb3de8fe7/Core/CFODQuery.c#L30)
- [CFODNode.c](https://github.com/darlinghq/darling-opendirectory/blob/c71415e716a8cba3e1539236b49de8abb3de8fe7/Core/CFODNode.c#L31)
- [CFODSession.c](https://github.com/darlinghq/darling-opendirectory/blob/c71415e716a8cba3e1539236b49de8abb3de8fe7/Core/CFODSession.c#L32)
- [CFODRecord.c](https://github.com/darlinghq/darling-opendirectory/blob/c71415e716a8cba3e1539236b49de8abb3de8fe7/Core/CFODRecord.c#L29)

**Comment by ahyattdev** [comment]
Looks good, I am going to restructure so there aren't two folders called `OpenDirectory` and `opendirectory`, I think the latter was created by me as a mistake a while ago.

**Pull request merged**